pkg/dab/find.go

```diff
@@ -165,35 +172,78 @@ func FindActionableFromFS(
 
 		// None of the filename peeks found a thing?
 		// Okay.  If we can search further up, let's do so.
+		foundPath = "" // clear found path
 		if !searchUp {
-			foundPath = ""
+			remainingSearchPath = filepath.Dir(remainingSearchPath)
 			return
 		}
-		remainingSearchPath = filepath.Dir(remainingSearchPath)
-		if remainingSearchPath == "/" || remainingSearchPath == "." {
+		if len(remainingSearchPath) == 1 && (remainingSearchPath == "/" || remainingSearchPath == "." || remainingSearchPath == "") {
 			return
 		}
+		remainingSearchPath = filepath.Dir(remainingSearchPath)
 	}
 }
 
-// findActionableFromFile peeks at the first few bytes of a stream
+// optionalModulePlotFromFile will load a module from file and an adjacent plot from file.
+// Does not return an error for non-existent files.
+func optionalModulePlotFromFile(fsys fs.FS, path string) (*wfapi.Module, *wfapi.Plot, error) {
+	m, err := ModuleFromFile(fsys, path)
+	if err != nil {
+		if serum.Code(err) != wfapi.ECodeMissing { // notexist is ignored.
+			// Any other error means our search has blind spots: error out.
+			return nil, nil, wfapi.ErrorSearchingFilesystem("loading module", err)
+		}
+		return nil, nil, nil
+	}
+	// A module may also have a plot next to it; load that eagerly too.
+	plotPath := filepath.Join(filepath.Dir(path), MagicFilename_Plot)
+	p, err := optionalPlotFromFile(fsys, plotPath)
+	if err != nil {
+		return m, nil, err
+	}
+	return m, p, nil
+}
+
+// optionalPlotFromFile will load a plot from file.
+// Does not return an error for non-existent files.
+func optionalPlotFromFile(fsys fs.FS, path string) (*wfapi.Plot, error) {
+	p, err := PlotFromFile(fsys, path)
+	if err != nil {
+		if serum.Code(err) != wfapi.ECodeMissing { // notexist is ignored.
+			// Any error that's just just notexist: means our search has blind spots: error out.
+			return nil, wfapi.ErrorSearchingFilesystem("modules, plots, or formulas", err)
+		}
+		return nil, nil
+	}
+	return p, nil
+}
+
+const (
+	guessMagic_ModuleV1 = "module.v1"
+	guessMagic_PlotV1   = "plot.v1"
+	guessMagic_Formula  = "formula" // REVIEW: not sure why we don't use FormulaCapsule V1 value here
```

Comment:
Because there's no such thing.  We chose to put a version-hinting capsule type around Formula and around FormulaContext separately... and thus the top level object in a file containing FormulaAndContext doesn't actually have any version hinting in it: those hints are in deeper structures.

Whether all that was a good idea or not...  I really don't know.  (It's not an accident; we did think about it at the time, in some depth.  But look at it again now: it certainly does look Weird.)

Reply:
There's still a `Capsule` type in the schema. https://github.com/warptools/warpforge/blob/master/wfapi/wfapi.ipldsch#L40-L43